### DISCUSSION TITLE
RNode - Heltec Lora32 V2 display?

### DISCUSSION THREAD

**Discussion by cobraPA** [markqvist/Reticulum#310] [Help & Questions] [answered]
It sounds like the display should work after doing an rnodeconf --autoinstall on this device?
Display was live at first boot with the brand new device.  auto install says success and I can check the device and even run loramon, but display is blank.

**Comment by Dant3s** [comment]
Hey!   I have 4 Heltec Lora32 v2 (Not the usb-c version) From all of them only one had that "issue". But the screen is randomly blank (aka shutdown)  and never stays blank, after a moment it starts. I eventually "fix" it (Since i don't really know why that happend) reflashing a few times the board.

**Comment by markqvist** [comment]
I just tested it, and can confirm there is a bug in firmware version `1.60` that makes display initialisation fail. This was introduced in https://github.com/markqvist/RNode_Firmware/commit/b12fc82a80f6d9dc40a4a2615cbfba586c8bc85f when support for the LilyGO T3 v1.0 boards were added, due to some faulty logic in the display init.

I am releasing a fix later today along with updated RNS packages :)

**Comment by markqvist** [comment, marked as answer]
Version `1.61` of the RNode Firmware is released now, and you should be able to update your Heltec devices with `rnodeconf` to restore display functionality :)

**Reply by cobraPA** [reply]
Wow, thanks Mark!

**Reply by cobraPA** [reply]
Updated, display is back.  Nice job. :)

**Comment by cobraPA** [comment]
Is there a change need for T-Beam boards also?  I just flashed a T-Beam V1.1 and the display is dead now.  --info shows it alive though.

**Reply by markqvist** [reply]
Was it working on a previous version on that piece of hardware, or was it never working before? The T-Beams have various different display pin mappings, and have not made options for that in the installer yet.

**Reply by cobraPA** [reply]
Oh, no, it's a new device, so this is first version I tried.  Just noticed that meshtastic was supplied and had screen working and then did not when installed with Rnode 1.61 yesterday.

**Reply by markqvist** [reply]
Okay, it is most likely just an issue of pin mappings and I2C address of the display. Can you post a few photos of the device and screen? And if possible, the pin numbers the display connect to?

**Reply by markqvist** [reply]
Also, if it is printed on the display somewhere, let me know if it uses address `0x3C` or `0x3D`.

**Reply by cobraPA** [reply]
Not much on the display.  It looks like https://www.aliexpress.us/item/2255800992363816.html?spm=a2g0o.detail.1000023.11.305f5cd0V837L4&gatewayAdapt=glo2usa4itemAdapt .  Which claims the display driver is SSD1306.  The pins used are 22 -> SCL on display, 21 -> SDA.

**Reply by cobraPA** [reply]
![IMG_1629](https://github.com/markqvist/Reticulum/assets/22970414/729f445e-65f1-423f-9a47-7529ae7bebc8)
The display doesn’t have much on the visible side, no driver chips visible.

**Reply by markqvist** [reply]
Thanks, looks like it's just raw I2C without any reset lines or anything. Since we don't know the address, I'll see if I can figure something neat out that can autodetect it.

**Reply by cobraPA** [reply]
Yes, looks like code that LilyGo references uses https://github.com/bitbank2/OneBitDisplay, which passes -1 as the address and autodetects the 0x3C or 0x3D (and driver chip as well...)  That's as far as I got so far in case that helps.

**Comment by Swissbandit** [comment]
i reflashed 2 lora 2.1 aka 1.6.1 devices.
i noticed there was some errors trying this and i tested on 2 different os and machines. 
if after a failed attempt after some time the screen goes out, the next flash tryout was working and the device is fine after that.

**Reply by markqvist** [reply]
Thanks for the info. I'll keep an eye on this.
